### PR TITLE
Sample code for long-press to save recording

### PR DESCRIPTION
**Pull request by AkazaRenn**
This is the sample code to trigger saving a clip if `share` button is held down for longer than 500 ms. However, this would disallow users to remap this button for other purpose per game. Also, even if they don't remap, a screenshot will still be saved by Steam every time we save a clip, but I personally am fine with that.

Your codes to capture with `steam` + `start` are commented out because I don't know what the best way is to integrate my change into them 🥲

I've tested it to be working with an Xbox Series controller, suppose it will be the same for PS and NS controllers. Please let me know if you need help adding UI elements like enabling it, adjusting holding time length, etc.

~~If you prefer using this method, maybe we could get rid of the current key-combination solution and use holding either `QAM` or `Share` to be compatible with both Deck and controllers. But I don't have a Steam Deck thus there's no way I can test QAM...~~

### CODE REVIEW

**Comment by safijari** [comment]
Interesting. I think it should be possible for the two shortcuts to coexist. Holding down QAM is an intriguing idea. If you do it too long it shows the shortcuts and if we do it too short we run the risk of triggering it when the user is lowering their screen brightness or something.

**Comment by AkazaRenn** [comment]
> Interesting. I think it should be possible for the two shortcuts to coexist. Holding down QAM is an intriguing idea. If you do it too long it shows the shortcuts and if we do it too short we run the risk of triggering it when the user is lowering their screen brightness or something.

I didn't know QAM can combine with other buttons to adjust brightness 😂 in that case holding QAM may not be a good idea. And given that Deck has already used a key combination for taking screenshots, it should be fine to use another for saving clips. I think we can ignore the QAM part.

**Comment by AkazaRenn** [comment]
Just came up with an idea to wait for the timeout instead of share button release, will update the code

**Comment by AkazaRenn** [comment]
Updated the code, realized that we don't need to be stuck with only one handler 😉

Steam + start combination doesn't work with my Xbox controller, maybe it's not equivalent to the Xbox button, but maybe you can give it a try.

I've also found a way to delete screenshots, so we can delete the most recent screenshot if we have saved a clip instead. I didn't add it because a toast would still happen since we have no way to stop Steam from doing so, deleting the screenshot might be more confusing to the users.

I can add some UI elements to adjust
1. Length of clip saved by hotkeys (1s to 600s)
2. Toggle holding share button to save
3. Hold the share button for how long to save a clip

File: src/index.tsx [hunk 1]
```diff
@@ -20,10 +20,105 @@ import {
...
+			this.scanBlocked = false;
+		}, timeout)	
+	}
+
+ 	abstract handleButtonInput(val: any[]) : any;
+}
+
+class CombinedButtons extends Button {
 	pressedAt: number = Date.now();
+	buttonVals: number[];
```
Comment (AkazaRenn) [inline src/index.tsx:43]: Should be able to use flipping bits as well

**Comment by AkazaRenn** [comment]
By the way does the recorder support taking a screenshot? I would like to override the one by Valve (delete it whenever taken and do it by myself) so I can decide where to save them.

**Comment by safijari** [comment]
> By the way does the recorder support taking a screenshot? I would like to override the one by Valve (delete it whenever taken and do it by myself) so I can decide where to save them.

No, but I have another plugin called Shotty which moves screenshots to the Pictures folder under properly named app folders. You can look into that.

**Comment by AkazaRenn** [comment]
> > By the way does the recorder support taking a screenshot? I would like to override the one by Valve (delete it whenever taken and do it by myself) so I can decide where to save them.
> 
> No, but I have another plugin called Shotty which moves screenshots to the Pictures folder under properly named app folders. You can look into that.

That's nice! Then I don't need to worry about screenshots anymore.

**Pull request closed without merging**